tests/stp_test.py

```diff
@@ -0,0 +1,703 @@
+        print("exit code {}".format(result.exit_code))
+        print("result code {}".format(result.output))
+        assert result.exit_code != 0
+        assert "STP hello timer must be in range 1-10" in result.output
+
+        result = runner.invoke(
+            config.config.commands["spanning-tree"].commands["vlan"].commands["forward_delay"],
+            ["100", "2"], obj=db)
+        print("exit code {}".format(result.exit_code))
+        print("result code {}".format(result.output))
+        assert result.exit_code != 0
+        assert "STP forward delay value must be in range 4-30" in result.output
+
+        result = runner.invoke(
+            config.config.commands["spanning-tree"].commands["vlan"].commands["forward_delay"],
+            ["100", "42"], obj=db)
+        print("exit code {}".format(result.exit_code))
+        print("result code {}".format(result.output))
+        assert result.exit_code != 0
+        assert "STP forward delay value must be in range 4-30" in result.output
+
+        result = runner.invoke(
+            config.config.commands["spanning-tree"].commands["vlan"].commands["max_age"],
+            ["100", "4"], obj=db)
+        print("exit code {}".format(result.exit_code))
+        print("result code {}".format(result.output))
+        assert result.exit_code != 0
+        assert "STP max age value must be in range 6-40" in result.output
+
+        result = runner.invoke(
+            config.config.commands["spanning-tree"].commands["vlan"].commands["max_age"],
+            ["100", "45"], obj=db)
+        print("exit code {}".format(result.exit_code))
+        print("result code {}".format(result.output))
+        assert result.exit_code != 0
+        assert "STP max age value must be in range 6-40" in result.output
+
+        result = runner.invoke(
+            config.config.commands["spanning-tree"].commands["vlan"].commands["forward_delay"],
+            ["100", "4"], obj=db)
+        print("exit code {}".format(result.exit_code))
+        print("result code {}".format(result.output))
+        assert result.exit_code != 0
+        assert "2*(forward_delay-1) >= max_age >= 2*(hello_time +1 )" in result.output
+
+        result = runner.invoke(
+            config.config.commands["spanning-tree"].commands["vlan"].commands["priority"], ["100", "70000"], obj=db)
+        print("exit code {}".format(result.exit_code))
+        print("result code {}".format(result.output))
+        assert result.exit_code != 0
+        assert "STP bridge priority must be in range 0-61440" in result.output
+
+        result = runner.invoke(
+            config.config.commands["spanning-tree"].commands["vlan"].commands["priority"], ["100", "8000"], obj=db)
+        print("exit code {}".format(result.exit_code))
+        print("result code {}".format(result.output))
+        assert result.exit_code != 0
+        assert "STP bridge priority must be multiple of 4096" in result.output
```

Comment:
This is a huge testcases with many asserts, any of them failing will break the test and interrupt upcoming assertion. Can we break them down into smaller test cases. Consider using https://docs.pytest.org/en/7.1.x/example/parametrize.html

Reply:
Done